elastalert/elastalert.py

```diff
@@ -686,6 +687,9 @@ def get_segment_size(self, rule):
 
     def get_query_key_value(self, rule, match):
         # get the value for the match's query_key (or none) to form the key used for the silence_cache.
+        # Flatline ruletype sets "key" instead of the actual query_key
+        if isinstance(rule['type'], FlatlineRule) and 'key' in match:
+            return match['key']
```

Comment:
get_named_key_value won't work for this. It takes a key name then looks up the value of that key in the rule. Passing 'query_key' might return rule['username'] for example. I will add a unicode( ) though.